docs/adr/0005-remote-settings-client.md

```diff
@@ -0,0 +1,112 @@
+# A remote-settings client for our mobile browsers.
+
+* Status: proposed
+* Deciders:
+    * csadilek for the mobile teams ✔️
+    * bwong for the remote-settings team
+    * mhammond for the application-services team ✔️
+    * [others?]
+* Date: 2022-12-16
+
+## Context and Problem Statement
+
+Mozilla’s mobile browsers have a requirement to access the remote settings service, but currently lack any libraries or tools which are suitable without some work.
+A concrete use case is the management of search engine configurations, which are stored in Remote Settings for Firefox Desktop, but shipped as individual files on our mobile browsers, requiring application releases for all changes.
+This document identifies the requirements, then identifies tools which already exist and are close to being suitable, then identifies all available options we can take, and outlines our decision.
+
+### Requirements
+The requirements are for a library which is able to access Mozilla’s Remote Settings service and return the results to our mobile browsers.
+This list of requirements is not exhaustive, but instead focuses on the requirements which will drive our decision making process.
+As such, it identifies the non-requirements first.
+
+### Not a requirement
+The following items all may have some degree of desirability, but they are not hard requirements for the initial version
+* While the https connection to the server must be validated, there is no requirement to verify the content received by the server - ie, there’s no requirement to check the signature of the body itself.
+* There’s no requirement to validate the result of the server conforms to a pre-defined schema - we trust the server data.
+* There’s no requirement to return strongly-typed data to the applications - returning a JSON string/object is suitable.
+* There’s no requirement to cache server responses to the file-system - if the app requests content, it’s fine for the library to always hit the server.
+* There’s no requirement for any kind of scheduling or awareness of network state - when we are requested for content, we do it immediately and return an appropriate error if it can not be fetched.
+* There’s no requirement to support publishing records, requesting reviews or providing approvals via this new library.
+* There’s no requirement to manage buckets, groups and collections via this new library.
```

Comment:
The desktop remote settings client also uses [webpush](https://github.com/mozilla-services/remote-settings-lambdas#sync_megaphone) to receive changes from the remote settings server, which the experimentation platform can take advantage for rapid enrollment.  Might be worth mentioning that capability as a non-requirement here?

Reply:
Good idea, thanks - 3c356141e9177b963fbaf31aa07e9196b343cc27